### ISSUE TITLE
uBlock Origin support problems.

### ISSUE THREAD

**Issue by buthed010203** [brave/adblock-rust#197]
I was messing around with adblock settings trying to see why it wasn't blocking pages despite them being included in the [default uBlock lists](https://github.com/brave/adblock-resources/blob/24ff263e3b72da113c216ea360d7458a901e4998/filter_lists/default.json#L25). I came to the conclusion that brave adblock doesnt properly interpret `||site^$doc` as `||site^$document` which ublock origin does do. This causes much of that badware list to become completely useless. Using `ublock.org` as an example, we can see that it is in the list as [`||ublock.org^$doc`](https://github.com/uBlockOrigin/uAssets/blob/cae5cea684d2a81599192aaba68827b14805a4a3/filters/badware.txt#L54) but visiting it in brave doesn't block it. Adding the list manually at `brave://adblock` makes no difference either so it isn't that. Adding `||ublock.org^$doc` as a custom filter also does not work. Adding `||ublock.org^$document` does actually block the page (so does `||ublock.org^` FWIW). If you try opening the page with ublock origin installed, it will show that it has blocked `||ublock.org^$document` because it automatically expands it. This issue probably exists in other forms but I don't really want to go looking for them right now.

**TL;DR:**
Brave adblock doesn't convert `||mysite.com^$doc` into `||mysite.com^$document` which causes pages to not be blocked despite them being included in default filter lists.

**Comment by antonok-edm** [comment]
@buthed010203 thanks for reporting! Not sure how that got missed, but now it should be supported as of https://github.com/brave/adblock-rust/commit/fca90991070998bfd0d902d0875c6822b83b4c1c :smiley:

**Comment by buthed010203** [comment]
Thanks for the quick fix 👍

**Comment by antonok-edm** [comment]
Yep, unfortunately that's a little more complicated to add in (we don't currently have `popup`, `inline-font`, or `inline-script` either). But you're right in that it's not very commonly used, so it's not as high of a priority.

I've added a mention for it in https://github.com/brave/adblock-rust/issues/1
